subsys/bluetooth/mesh/shell/shell_ctl_cli.c

```diff
@@ -42,11 +42,17 @@ static int cmd_ctl_get(const struct shell *shell, size_t argc, char *argv[])
 
 static int ctl_set(const struct shell *shell, size_t argc, char *argv[], bool acked)
 {
-	uint16_t light = (uint16_t)strtol(argv[1], NULL, 0);
-	uint16_t temp = (uint16_t)strtol(argv[2], NULL, 0);
-	int16_t delta = (int16_t)strtol(argv[3], NULL, 0);
-	uint32_t time = (argc >= 5) ? (uint32_t)strtol(argv[4], NULL, 0) : 0;
-	uint32_t delay = (argc == 6) ? (uint32_t)strtol(argv[5], NULL, 0) : 0;
+	int err = 0;
+	uint16_t light = shell_strtoul(argv[1], 10, &err);
+	uint16_t temp = shell_strtoul(argv[2], 10, &err);
+	int16_t delta = shell_strtol(argv[3], 10, &err);
+	uint32_t time = (argc >= 5) ? shell_strtoul(argv[4], 10, &err) : 0;
+	uint32_t delay = (argc == 6) ? shell_strtoul(argv[5], 10, &err) : 0;
+
+	if (err) {
+		shell_warn(shell, "Unable to parse input string arg");
```

Comment:
Why is is used shell_warn here when it is a error. Should it be shell_error instead?